bounded/src/set.rs

```diff
@@ -285,148 +305,122 @@ fn translate<'a>(
-            .map(|term| {
-                term.get_and()
-                    .into_iter()
-                    .map(|term| term.get_guard(&indices))
-                    .collect::<Result<Vec<_>, _>>()
-            })
-            .collect::<Result<Vec<_>, _>>()
-    };
+    let d = extract(module).map_err(TranslationError::ExtractionError)?;
 
-    // inits should just be guards at this point
-    let inits: Vec<Vec<Guard>> = get_guards_from_dnf(inits)?;
+    let formula = |term| term_to_formula(&normalize(term, &context), &context, &HashMap::default());
 
-    // change inits from guards over the state space to states
-    let inits: Vec<BoundedState> = inits
-        .into_iter()
-        .flat_map(|conjunction| {
-            // compute all the constrained elements by adding them to a single state
-            let mut init = BoundedState(BitArray::ZERO);
-            let mut constrained = btreeset![];
-            for guard in &conjunction {
-                init.0.set(guard.index, guard.value);
-                constrained.insert(guard.index);
-            }
-
-            // compute all the unconstrained elements by doubling the number of states each time
-            let mut inits = vec![init];
-            for index in 0..indices.len() {
-                if !constrained.contains(&index) {
-                    inits = inits
-                        .into_iter()
-                        .flat_map(|state| {
-                            let mut with_unconstrained = state.clone();
-                            with_unconstrained.0.set(index, true);
-                            vec![state, with_unconstrained]
-                        })
-                        .collect();
-                }
+    // get cube
+    let mut init = BoundedState::ZERO;
+    let mut constrained = HashSet::default();
+    let inits = formula(Term::and(&d.inits))?;
+    for constraint in inits.get_and() {
+        if let Formula::Guard(Guard { index, value }) = constraint {
+            init.set(index, value);
+            constrained.insert(index);
+        } else {
+            return Err(TranslationError::InitNotConj);
+        }
+    }
+    // enumerate cube
+    let mut inits = vec![init];
+    for index in 0..context.indices.len() {
+        if !constrained.contains(&index) {
+            let mut with_unconstrained = inits.clone();
+            for init in &mut with_unconstrained {
+                init.set(index, true);
             }
-            inits
-        })
+            inits.append(&mut with_unconstrained);
+        }
+    }
+    // filter by axioms
+    let axiom = formula(Term::and(&d.axioms))?;
+    let inits = inits
+        .into_iter()
+        .filter(|init| axiom.evaluate(init))
         .collect();
 
-    let mut unconstrained_delta = 0;
-    let trs: Vec<Transition> = trs
-        .get_or()
+    // compute imperative transitions
+    assert!(d
+        .mutable_axioms(&module.signature.relations)
+        .next()
+        .is_none());
```

Comment:
Can you comment or add a message to this assert so that people who are not you will know that this is not supported, and does not signify a programming error?